app/bundles/ApiBundle/Tests/ApiPlatform/EventListener/MauticDenyAccessListenerTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\ApiBundle\Tests\ApiPlatform\EventListener;
+
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
+use ApiPlatform\Metadata\Resource\ResourceMetadataCollection;
+use Mautic\ApiBundle\ApiPlatform\EventListener\MauticDenyAccessListener;
+use Mautic\CoreBundle\Entity\FormEntity;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+final class MauticDenyAccessListenerTest extends TestCase
+{
+    /**
+     * @var MockObject|CorePermissions
+     */
+    private $corePermissionsMock;
+
+    /**
+     * @var ApiResource
+     */
+    private $resourceMetadata;
+
+    /**
+     * @var MockObject|ResourceMetadataCollectionFactoryInterface
+     */
+    private $resourceMetadataFactoryMock;
+
+    /**
+     * @var MockObject|RequestEvent
+     */
+    private $eventMock;
+
+    /**
+     * @var MauticDenyAccessListener
+     */
+    private $mauticDenyAccessListener;
+
+    protected function setUp(): void
+    {
+        $attributes = [
+            '_api_resource_class'      => 'TestClass',
+            '_api_operation_name'      => 'Test',
+            'item_operation_name'      => 'Test',
+        ];
+        $parameterBagMock = $this->createMock(ParameterBag::class);
+        $parameterBagMock
+            ->expects($this->exactly(1))
+            ->method('all')
+            ->willReturn($attributes);
+        $formEntityMock = $this->createMock(FormEntity::class);
+        $formEntityMock
+            ->expects($this->atMost(1))
+            ->method('getCreatedBy')
+            ->willReturn(0);
+        $parameterBagMock
+            ->expects($this->exactly(1))
+            ->method('get')
+            ->with('data')
+            ->willReturn($formEntityMock);
+        $requestMock                       = $this->createMock(Request::class);
+        $requestMock->attributes           = $parameterBagMock;
+        $this->corePermissionsMock         = $this->createMock(CorePermissions::class);
+        $this->resourceMetadataFactoryMock = $this->createMock(ResourceMetadataCollectionFactoryInterface::class);
+        $this->eventMock                   = $this->createMock(RequestEvent::class);
+        $this->eventMock
+            ->expects($this->exactly(1))
+            ->method('getRequest')
+            ->willReturn($requestMock);
```

Comment:
Let's not mock DTO objects with no dependencies on other PHP services.
```suggestion
        $this->requestEvent = new RequestEvent();
        $this->requestEvent->setRequest($requestMock);
```

Reply:
RequestEvent class extends KernelEvent which has dependencies. Phpunit still complains if I instantiate it this way

Reply:
I see. Let's keep the mock then

Reply:
Okay I found that I am able to use this method to instantiate using some simple modifications.